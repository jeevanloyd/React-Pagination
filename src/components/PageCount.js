import React from "react";

export const PageCount = (props) => {
    return (

        <div key={props.limit} className="input-group mb-3" style={{ width: "auto" }}>
            <select
                className="form-select"
                id="pageselect"
                aria-label="Default select example"
                value={props.limit}
                onChange={(e) => props.setLimit(e.target.value)}
            >
                <option value="4">4 per Page</option>
                <option value="8" >
                    8 per Page
                </option>
                <option value="16">16 per Page</option>
                <option value="32">32 per Page</option>
                <option value="64">64 per Page</option>
            </select>
        </div>

    );
};
