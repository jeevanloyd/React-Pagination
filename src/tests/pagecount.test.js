import React from 'react';
import { shallow } from 'enzyme';
import PageCount from '../components/PageCount';


test('should test Header component', () => {
    const wrapper = shallow(<PageCount />);
    expect(wrapper).toMatchSnapshot();
});