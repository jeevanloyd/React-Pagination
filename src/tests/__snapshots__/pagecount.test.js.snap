// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Header component 1`] = `
<div
  className="input-group mb-3"
  style={
    Object {
      "width": "auto",
    }
  }
>
  <select
    aria-label="Default select example"
    className="form-select"
    id="pageselect"
    onChange={[Function]}
  >
    <option
      value="4"
    >
      4 per Page
    </option>
    <option
      value="8"
    >
      8 per Page
    </option>
    <option
      value="16"
    >
      16 per Page
    </option>
    <option
      value="32"
    >
      32 per Page
    </option>
    <option
      value="64"
    >
      64 per Page
    </option>
  </select>
</div>
`;
